content/articles/firebase-vue-authentication/index.md

```diff
@@ -0,0 +1,437 @@
+### Managing User State in Vue.js and Firebase Applications
+In this article, we will take a look at how we can use Vuex 4 together with the Composition API in Vue 3.0 to set up a state management system for authenticating users with Firebase Auth. This will help us to have a global state that we can access from any component in the entire Vue application.
+
+### Pre-requisites
+- Basic understanding of Vue.js and the Composition API.
+- Familiarity with Firebase version 9.
+- Node.js installed on your local machine.
+- A web browser.
+
+### Goals
+- Working with the Vuex 4 and the Composition API.
+- Managing user state in Vue web applications.
+- Integrating Vue.js with Firebase
+
+### Getting Started
+To create a Vue.js app, you need to have the Vue CLI. To install it, run this command in your terminal:
+```bash
+npm install -g @vue/cli
+```
+After the installation is complete, go ahead and create a Vue app with this command:
+```bash
+vue create firebase-vue-user-management
+```
+On this screen, choose `Vue 3`:
+
+![preset](./section-engineering/firebase-vue-authentication/preset.png)
+
+And lastly on this screen choose `Use NPM`:
+
+![package manager](./section-engineering/firebase-vue-authentication/packagemanager.png)
+
+The above configurations will create a Vue 3 application with npm as the package manager.
+
+Now, run these commands to install the dependencies that we will use in this project:
+```bash
+npm install vue-router@4
+
+npm install vuex@next --save
+```
+- vue-router@4 - To set up routing in the Vue app.
+- vuex@next - For state management.
+  
+### Setting up Firebase Auth
+To create a new firebase project, click [here](https://firebase.google.com/) and select `Go to console` on the top navigation bar.
+
+Here, click on `Add new project` and name the project `vue-firebase-auth`.
+![New Project](./section-engineering/firebase-vue-authentication/addnewproject.png)
+
+You will be redirected to the project dashboard. On the dashboard, click on the web icon to register your front-end app.
+![Dashboard](./section-engineering/firebase-vue-authentication/dashboard.png) 
+Name your app `vuex-firebase-authentication` and click on `Register app`.
+
+Copy the configuration provided by Firebase and click `Continue to console`.
+
+Back on the dashboard, click `Authentication` in the left sidebar.
+![Authentication](./section-engineering/firebase-vue-authentication/authentication.png)
+
+Enable `Email/Password` and save.
+
+### Adding Firebase to the Vue App
+Open the terminal and type:
+```bash
+npm install firebase
+```
+This will install firebase.
+
+After the installation is complete, create a new folder in the `src` folder and name it `firebase`. Inside this folder, create a file `config.js`. Modify the file with the following code:
+```js
+//Import the required methods
+import { inititializeApp } from 'firebase/app'
```

Comment:
The word `initialize` is spelled wrong and that is what was causing my app to not render.

```suggestion
import { initializeApp } from 'firebase/app'
```